ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -12,15 +12,21 @@ const TRACE_FUNC_CTRLP_ARGIDX: u16 = 0;
 /// A TraceBuilder frame. Keeps track of inlined calls and stores information about the last
 /// processed stackmap, call instruction and its arguments.
 struct Frame<'a> {
+    // Index of the function of this frame.
+    func_idx: Option<FuncIdx>,
```

Comment:
What would `None` mean?

Reply:
Not initialised. This is an unfortunate side-effect of not knowing the function index of the first block at this time. It only affects the first frame. I made it an option so that we get a crash if we forget to update the field here https://github.com/ykjit/yk/pull/1121/files#diff-0019b0cfa54cfb8c953a80e8ab4e0ce900b5fada3100f573c59c09feadd32fc6R665

Reply:
Yeah nice one. Maybe just add "This is only None when ... "?